app/topics/topic-management/add-existing-subtopic/add-existing-subtopic.component.ts

```diff
@@ -36,12 +34,24 @@ export class AddExistingSubtopicComponent {
     this.showChange.emit(this.show);
   }
 
+  private filterTopics() {
+    for(var subtopic of this.subtopics) {
+      for(var searchResult of this.searchResults) {
```

Comment:
don't use `var` for variable declaration. that is Javascript syntax. We are using Typescript so use `let` instead.
In the inner loop you have to use an index because you are using `splice` which expect the index of the element as first argument.
So the function should look like this:
```
private filterTopics() {
    for (let subtopic of this.subtopics) {
      for (let i = 0; i < this.searchResults.length; i++) {
        if (subtopic.id === this.searchResults[i].id || this.parent.id === this.searchResult[i].id) {
          this.searchResults.splice(i, 1);
          break;
        }
      }
    }
  }
```